### PR TITLE
Python SDK Mutation and Ext Event

### PR DESCRIPTION
**Pull request by sauljabin**
* Add wait_for_event
* Add mutate
* Add NodeOutput
* Add utils
* Fix has_context function

### CODE REVIEW

File: sdk-python/littlehorse/user_utils.py [hunk 1]
```diff
@@ -100,3 +100,24 @@ def register_task(
             logging.info(f"TaskDef {name} already exits, skipping")
             return
         raise e
+
+
+def register_external_event(
```
Comment (coltmcnealy-lh) [inline sdk-python/littlehorse/user_utils.py:105]: This is nice! Java and Go don't have it yet :wink:

Comment (coltmcnealy-lh) [reply sdk-python/littlehorse/user_utils.py:105]: Actually, could you please rename it to `register_external_event_def`? The `def` is important, `ExternalEvent != ExternalEventDef`.

**Review by coltmcnealy-lh — APPROVED** [review]
:pray:

File: sdk-python/littlehorse/user_utils.py (hunk 1 above)
Comment (sauljabin) [reply sdk-python/littlehorse/user_utils.py:105]: is it possible "register" a event? isn't like post_external_event? 

register_external_event is for PutExternalEventDef
post_external_event is PutExternalEvent

or is it confuse? 🤔

Comment (sauljabin) [reply sdk-python/littlehorse/user_utils.py:105]: in lhctl we have postEvent

Comment (coltmcnealy-lh) [reply sdk-python/littlehorse/user_utils.py:105]: This method creates external event definition, so it should have _def_ at the end.

You "postEvent` an External Event, which is dfferent than `externalEvent`.

It's `lhctl deploy exteranlEventDef` and `lhctl postEvent`... probably shouldbe `lhctl put externalEvent` but oops

Comment (sauljabin) [reply sdk-python/littlehorse/user_utils.py:105]: what is instead of register we use "create"

**Pull request merged**